src/test/scala/utils/TestUtils.scala

```diff
@@ -84,4 +84,12 @@ object TestUtils {
     val randBasis: RandBasis = new RandBasis(new ThreadLocalRandomGenerator(new MersenneTwister(seed)))
     DenseMatrix.rand(numRows, numCols, Gaussian(0.0, 1.0)(randBasis))
   }
+
+  /* Convert a BGR image (channels x (rows x cols)) to RGB */
```

Comment:
Can you add a quick note on the format of this matrix since it's not an `Image` - alternatively maybe it should just be implemented in terms of the `Image` API